Source/Diagnostics/ComputeDiagFunctors/RhoFunctor.cpp

```diff
@@ -15,7 +19,22 @@ RhoFunctor::operator() ( amrex::MultiFab& mf_dst, const int dcomp, const int /*i
     auto& mypc  = warpx.GetPartContainer();
 
     // Deposit charge density
-    std::unique_ptr<amrex::MultiFab> rho = mypc.GetChargeDensity( m_lev );
+    // Call this with local=true since the parallel transfers will be handled
+    // by ApplyFilterandSumBoundaryRho
+    std::unique_ptr<amrex::MultiFab> rho = mypc.GetChargeDensity( m_lev, true );
+
+    // Call this in case filtering is turned on
+    warpx.ApplyFilterandSumBoundaryRho(m_lev, m_lev, *rho, 0, rho->nComp());
+
+#ifdef WARPX_DIM_RZ
+    using Idx = SpectralAvgFieldIndex;
+    if (WarpX::use_kspace_filter) {
+        auto & solver = warpx.get_spectral_solver_fp(m_lev);
```

Comment:
It seems that this can be replaced by
```suggestion
        auto solver = spectral_solver_fp[m_lev];
        solver->ForwardTransform(*rho, Idx::rho_new);
        solver->ApplyFilter(Idx::rho_new);
        solver->BackwardTransform(*rho, Idx::rho_new);
```
Therefore, I am not sure whether it is really necessary to define a new function `get_spectral_solver_fp`. Was there another motivation for this new function?

Reply:
spectral_solver_fp is defined only in the `WarpX` class.